File: Leetcode/83-Remove-Duplicates-from-Sorted-List/RemoveDuplicatesFromSortedList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if (head == NULL || head->next == NULL) {
            return head;
        }
        ListNode* cur = head;
        ListNode* nextVal;
        while (cur!=NULL && cur->next!=NULL) {
            nextVal = cur->next;
            while (nextVal!=NULL && nextVal->val == cur->val) {
                nextVal = nextVal->next;
            }
            cur->next = nextVal;
            cur = cur->next;
        }
        return head;
    }
};

File: Leetcode/83-Remove-Duplicates-from-Sorted-List/RemoveDuplicatesFromSortedList.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head == None or head.next == None:
            return head
        
        cur = head;
        while cur != None and cur.next != None:
            nextVal = cur.next
            while nextVal != None and nextVal.val == cur.val :
                nextVal = nextVal.next
            cur.next = nextVal
            cur = cur.next
        return head
        
